dask_expr/_expr.py

```diff
@@ -2923,6 +2838,62 @@ def _execute_task(graph, name, *deps):
         return dask.core.get(graph, name)
 
 
+def collect_depdendents(expr) -> defaultdict:
```

Comment:
This is currently not working as it should if you do dask.compute(a,b,c), but neither does the status quo.

The tuple expression is what will address this problem

> This will only return the correct result if expr is a leaf, doesn't it?

depends on what you are looking for, it would be correct if you are only interested in a subtree, otherwise not

Reply:
> depends on what you are looking for, it would be correct if you are only interested in a subtree, otherwise not

Well, we're not always dealing with trees, are we? Consider

```mermaid
graph BT;
    A1-->B1;
    A2-->B1;
    A3-->B2;
    A4-->B2;
    B1-->C1;
    B2-->C1;
    B2-->C2;
    C1-->D;
    C2 -->D

```

Unless you start with `D` you will not get the dependents of `B2` right since you're not walking the entire graph.

I _assume_ this is fine since we are always starting at a terminal node?

Reply:
Sorry I wasn't clear enough.

We are always starting with the node that is the leaf node for the thing that we want to compute, so yes